### PR TITLE
feat: Built the EsLint Plugin for million

### PR DESCRIPTION
**Pull request by Drex72**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [x] Code changes have been tested against prettier, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the codebase
  - [] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
  - [x] This PR changes the internal workings with no modifications to the external API (bug fixes, performance improvements)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

Fixes issue #394 
/claim https://github.com/aidenybai/million/issues/394

This is a really amazing project @aidenybai, would love to take on more tasks later after this. If that's allowed :)

Thanks for the opportunity to contribute!

### CODE REVIEW

File: packages/eslint-plugin-millionjs/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# eslint-plugin-millionjs-drex
```
Comment (aidenybai) [inline packages/eslint-plugin-millionjs/README.md:1]: Can you change this to eslint-plugin-million. I can try publishing it on my side. Make sure to change the rest of the instances as well.

**Review by aidenybai — CHANGES_REQUESTED** [review]
just one nit.

File: packages/eslint-plugin-millionjs/README.md (hunk 1 above)
Comment (Drex72) [reply packages/eslint-plugin-millionjs/README.md:1]: No problem
I just put it there so everyonbe can install it from the npm registry and test
i'd do just that

**Comment by Drex72** [comment]
> just one nit.

Just Resolved it Now

File: packages/eslint-plugin-millionjs/docs/rules/avoid-non-deterministic-returns.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+const ComponentBlock = block(Component);
+```
+
+Examples of **correct** code for this rule:
+
+````js
+function Component() {
+  const [count, setCount] = useState(initial.count);
+ 
+
```
Comment (nucleartux) [inline packages/eslint-plugin-millionjs/docs/rules/avoid-non-deterministic-returns.md:36]: Something is wrong here

File: packages/eslint-plugin-millionjs/docs/rules/avoid-non-deterministic-returns.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Examples of **correct** code for this rule:
+
+````js
+function Component() {
+  const [count, setCount] = useState(initial.count);
+ 
+ 
+  return count > 5 ? (
+    'Count is greater than 5'
+  )
```
Comment (nucleartux) [inline packages/eslint-plugin-millionjs/docs/rules/avoid-non-deterministic-returns.md:39]: Something is wrong here

Comment (Drex72) [reply packages/eslint-plugin-millionjs/docs/rules/avoid-non-deterministic-returns.md:39]: as how
could you please elaborate
I would really appreciate that

Comment (Drex72) [reply packages/eslint-plugin-millionjs/docs/rules/avoid-non-deterministic-returns.md:39]: ohhh
i see now
do you mean no closing for the ternary?

Comment (nucleartux) [reply packages/eslint-plugin-millionjs/docs/rules/avoid-non-deterministic-returns.md:39]: Yep :)

Comment (Drex72) [reply packages/eslint-plugin-millionjs/docs/rules/avoid-non-deterministic-returns.md:39]: I probably just forgot ..😅😅 thanks

**Comment by tobySolutions** [comment]
Any updates on this yet?

**Comment by tobySolutions** [comment]
Any updates on this @aidenybai, @Drex72?

**Pull request closed without merging**